cli_tools/google-osconfig-agent/ospatch/patch_run.go

```diff
@@ -224,35 +217,34 @@ func (r *patchRun) rebootIfNeeded(ctx context.Context, postUpdate bool) (shouldS
 		}
 	}
 
-	if reboot {
-		if r.reportState(ctx, osconfigpb.Instance_REBOOTING) {
-			return true
-		}
+	if !reboot {
+		return false
+	}
 
-		r.PatchStep = postPatchRebooted
-		r.saveState()
+	if r.reportState(osconfigpb.Instance_REBOOTING) {
+		return true
+	}
 
-		if r.Job.DryRun {
-			logger.Infof("Dry run - not rebooting for patch job '%s'", r.Job.PatchJobName)
-		} else {
-			err := rebootSystem()
-			if err != nil {
-				r.finishAndReportError(ctx, fmt.Sprintf("Failed to reboot system: %v", err))
-				return true
-			}
+	if r.Job.DryRun {
+		logger.Infof("Dry run - not rebooting for patch job '%s'", r.Job.PatchJobName)
+		return false
+	}
 
-			// Reboot can take a bit, shutdown the agent so other activities don't start.
-			os.Exit(0)
-			return true
-		}
+	r.RebootCount++
+	if err := rebootSystem(); err != nil {
+		r.finishAndReportError(fmt.Sprintf("Failed to reboot system: %v", err))
+		return true
 	}
-	return false
+
+	// Reboot can take a bit, shutdown the agent so other activities don't start.
```

Comment:
The agent being a service, will it be automatically restarted?

Reply:
I'll have to do some testing here, killing the process is probably no the best thing to do. Changed this to a loop that sleeps forever. i may want to get fancier later and time out eventually but for now this should work.